### PR TITLE
Fixes bodies responding to phobias

### LINKED ISSUE #36118: Phobias can cause ghosts to scream
Round ID: 84447

Phobias can still affect ghosts that are inhabiting their own corpses, which includes giving phobia messages and causing screaming.
In my case, it was with a robot phobia, but I expect it applies to others.

To reproduce: Get a robot phobia, die, stay in your corpse (i.e. don't become a free-moving ghost), then have AIs speak, or robots move past, etc. One of the random phobia effects you'll get is a scream which is spoken into dead chat.

### PR DESCRIPTION
**Pull request by ghost**
When a dead body percieves a phobia trigger, they no longer freak out.

:cl:
fix: Dead bodies no longer freak out about phobias
/:cl:

Fixes #36118

### CODE REVIEW

File: code/datums/brain_damage/phobia.dm [hunk 1]
```diff
@@ -85,31 +85,32 @@
...
-			owner.Jitter(5)
-			owner.blind_eyes(10)
-		if(4)
-			owner.dizziness += 10
-			owner.confused += 10
-			owner.Jitter(10)
-			owner.stuttering += 10
\ No newline at end of file
+	if(owner.stat != DEAD)
+		var/message = pick("spooks you to the bone", "shakes you up", "terrifies you", "sends you into a panic", "sends chills down your spine")
```
Comment (SpaceManiac) [inline code/datums/brain_damage/phobia.dm]: Prefer an early return:
```dm
    if(owner.stat == DEAD)
        return
    ...rest of proc...
```

Comment (Singularbyte) [reply code/datums/brain_damage/phobia.dm]: Thanks. Made the change.

**Comment by Singularbyte** [comment]
I'm not exactly sure of the best way to do that to be honest. It would be easy enough to apply it to the on_hear of phobias explicitly and in other specific traumas, but to generalise it I'd need a decent idea of what it would do differently if the conditions were met. The only thing I can think of would be a new on_actually_hear proc, but that might be a bit redundant?

**Comment by XDTM** [comment]
put the check in the brain_traum/proc/on_heal, and call if(!..()) return in the first line of all the subtypes that use it

**Pull request merged**